lib/request/complement.js

```diff
@@ -54,30 +38,9 @@ module.exports = function complement(requested, optimized, depthDifferences, tre
     for (var i = 0, len = optimized.length; i < len; ++i) {
         var oPath = optimized[i];
         var rPath = requested[i];
-        var depthDiff = depthDifferences[i];
         var subTree = tree[oPath.length];
 
-        // no deduping possible
-        if (!subTree) {
-            optimizedComplement[++complementLength] = oPath;
-            requestedComplement[complementLength] = rPath;
-            continue;
-        }
-        // fully deduped
-        if (hasIntersection(subTree, oPath, 0)) {
```

Comment:
@jcranendonk is there any performance difference between running `hasIntersection` and `findPartialIntersections`? Might very well depend on their relative usage in apps..

Reply:
Source here: https://github.com/Netflix/falcor-path-utils/blob/master/lib/hasIntersection.js

The complexity of `hasIntersection` and `findPartialIntersections` looks essentially the same to me. The only difference is that the latter builds up the intersection/complement arrays, and updates the current path on recursion. In the case where `hasIntersection` would return true, I think exactly the same work is done.